src/shisen_cpp/camera/node/camera_node.cpp

```diff
@@ -69,6 +71,34 @@ void CameraNode::on_camera_config(int width, int height)
   camera_config_publisher->publish(camera_config_provider->get_camera_config());
 }
 
+void CameraNode::save_image(cv::Mat mat)
+{
+  struct stat info;
+
+  if (stat("image", &info) != 0) {
+    if (mkdir("image", 0777) == -1) {
+      RCLCPP_ERROR(node->get_logger(), "Error creating `image` directory!");
+      return;
+    }
+  }
```

Comment:
Use [`std::filesystem::exists`](https://en.cppreference.com/w/cpp/filesystem/exists) and [`std::filesystem::create_directory`](https://en.cppreference.com/w/cpp/filesystem/create_directory) instead:
```suggestion
  if (!std::filesystem::exists("image")) {
    if (!std::filesystem::create_directory("image")) {
      RCLCPP_ERROR(node->get_logger(), "Error creating `image` directory!");
      return;
    }
  }
```